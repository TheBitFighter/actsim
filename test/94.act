deftype info (int x[4]) { }

function func (info i) : info
{
  chp {
    self := i;
    self.x[0] := self.x[0] + 1
  }
}

defproc sink(chan?(info) X)
{
  info i;

  chp {
   *[ X?i; log ("got: ", i.x[0], " ", i.x[1], " ", i.x[2], " ", i.x[3]);
      i := func(i);
      log ("got2: ", i.x[0], " ", i.x[1], " ", i.x[2], " ", i.x[3])
    ]
  }
}

defproc test()
{
   info x;
   chan(info) y;

   sink s(y);

   chp {
     x.x[0] := 34;
     x.x[1] := 3;
     x.x[2] := 17;
     x.x[3] := 1;
     log ("x is ", x.x[0]);
     y!x
   }
}
