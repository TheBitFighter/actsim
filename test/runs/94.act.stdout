[                  40] <>  x is 34
[                  50] <s>  got: 34 3 17 1
[                  70] <s>  got2: 35 3 17 1
